packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -469,6 +469,14 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
                 this.ops = ops;
                 return id ? [{ id, treeId: undefined! }] : [];
+            }).catch(async (error) => {
+                const errorType = error.errorType;
+                // Clear the cache on 401/403/404 on snapshot fetch from network because this means either the user doesn't have permissions
+                // permissions for the file or it was deleted. So the user will again try to fetch from cache on any failure in future.
+                if (this.hostPolicy.concurrentSnapshotFetch && (errorType === DriverErrorType.authorizationError || errorType === DriverErrorType.fileNotFoundOrAccessDeniedError)) {
```

Comment:
I'm not sure checking for this.hostPolicy.concurrentSnapshotFetch is needed here.

Reply:
Yeah it does not seem that important.